### PR TITLE
feat: add settings screen

### PR DESCRIPTION
**Pull request by dnlggr**
Adds a simple settings screen and moves the "show balance" and "display amounts in sats" switches there. Nothing fancy yet because I wanted to keep the pull request as small as possible. We can do UI improvements later.

The main addition is a [React Context](https://reactjs.org/docs/context.html) that persists settings in local storage and keeps those settings in sync across the app. 

## Read Settings in a Component

``` js
import { useSettings } from '../context/SettingsContext'

/* later in the component */

const settings = useSettings() // e.g. { unit: "sats", showBalance: false }

<p>Unit: {settings.unit}</p> // read settings; when they change this component will be re-rendered
```

## Update Settings from a Component

``` js
import { useSettingsDispatch } from '../context/SettingsContext'

/* later in the component */

const settingsDispatch = useSettingsDispatch()

settingsDispatch({ unit: "BTC" }) // update that value of the settings; persists them in local storage and re-rendering all components that depend on them
```

Maybe we want to have more fine granular settings instead of having everything in one big JSON in local storage. Then we could have more control over what components get re-rendered when a single setting changes. But for now I feel like it is good enough.

## 📸 

Screenshot 2022-01-27 at 16 57 07
Screenshot 2022-01-27 at 16 57 14

### CODE REVIEW

File: src/components/App.jsx [hunk 1]
```diff
@@ -158,12 +159,12 @@ export default function App() {
                 <rb.Offcanvas.Body>{nav}</rb.Offcanvas.Body>
               </rb.Navbar.Offcanvas>
               {currentWallet && (
-                <rb.Nav className="ms-sm-auto ms-lg-0 d-block order-sm-0">
-                  <Link to="/wallet" className="nav-link d-inline-block">
+                <rb.Nav className="ms-sm-auto ms-lg-0 d-inline-flex flex-row order-sm-0">
```
Comment (dnlggr) [inline src/components/App.jsx:162]: Had to change this to `inline-flex` to vertically center the gear icon. I couldn't find another way but I feel like I just missed something obvious.

File: src/components/App.jsx [hunk 2]
```diff
@@ -181,22 +182,38 @@ export default function App() {
...
+                    </svg>
```
[the remark this answers is not in the document]
Comment (dnlggr) [reply src/components/App.jsx:215]: Makes sense! Once that #10 is merged I'll update this. 👍

[the remark this answers is not in the document]
Comment (GBKS) [reply src/components/App.jsx:215]: You can always use https://bitcoinicons.com

**Review by dergigi — APPROVED** [review]
Looks good to me ✅ and everything worked as expected when I tested it.

**Review by dennisreimann — APPROVED** [review]

File: src/components/App.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (dnlggr) [reply src/components/App.jsx:215]: Oh awesome! Definitely gonna do that. 😍

**Pull request merged**